cassandane/Cassandane/PortManager.pm

```diff
@@ -74,6 +78,31 @@ sub alloc
     die "No ports remaining";
 }
 
+sub port_is_free
+{
+    my $host = shift;
+    my $port = shift;
+
+    # If we can bind to the port no one else is currently using it
+    my $socket = IO::Socket::IP->new(
+        LocalAddr => $host,
+        LocalPort => $port,
+        Proto     => 'tcp',
+        ReuseAddr => 1,
+    );
```

Comment:
Oh, I guess this gets unbound again automatically when `$socket` goes out of scope as `port_is_free()` returns.  I was staring at it for a while wondering how binding the port here wouldn't create the exact problem we were trying to avoid...  Spend a lot of time writing C, forget other languages have destructors. 😅

Reply:
:)